module/zfs/dsl_bookmark.c

```diff
@@ -377,7 +500,78 @@ dsl_bookmark_create_sync_impl(const char *bookmark, const char *snapshot,
 	dsl_dataset_rele(snapds, FTAG);
 }
 
+
 static void
+dsl_bookmark_create_sync_impl_book(
+    const char *new_name, const char *source_name, dmu_tx_t *tx)
+{
+	dsl_pool_t *dp = dmu_tx_pool(tx);
+	dsl_dataset_t *bmark_fs_source, *bmark_fs_new;
+	char *source_shortname, *new_shortname;
+	zfs_bookmark_phys_t source_phys;
+
+	VERIFY0(dsl_bookmark_hold_ds(dp, source_name, &bmark_fs_source, FTAG,
+	    &source_shortname));
+	VERIFY0(dsl_bookmark_hold_ds(dp, new_name, &bmark_fs_new, FTAG,
+	    &new_shortname));
+	/* Bookmarks must be on the same dataset */
+	VERIFY3U(bmark_fs_source->ds_object, ==, bmark_fs_new->ds_object);
```

Comment:
I think you should be able to create a bookmark from a bookmark in the origin, just like you can create a bookmark from a snapshot in the origin.

Reply:
Yes, I already addressed this in the `*_check` but forgot to adjust the `*_sync`. Should I do the `dsl_dataset_is_before` check again in a VERIFY or just drop it?

Reply:
... dropping it